manim/mobject/types/vectorized_mobject.py

```diff
@@ -1294,14 +1295,14 @@ def gen_cubic_bezier_tuples_from_points(
         # Basically take every nppcc element.
         return tuple(points[i : i + nppcc] for i in range(0, len(points), nppcc))
 
-    def get_cubic_bezier_tuples(self) -> npt.NDArray[Point3D_Array]:
+    def get_cubic_bezier_tuples(self) -> CubicBezierPoints_Array:
         return self.get_cubic_bezier_tuples_from_points(self.points)
 
     def _gen_subpaths_from_points(
         self,
-        points: Point3D_Array,
+        points: CubicBezierPath,
         filter_func: Callable[[int], bool],
-    ) -> Generator[Point3D_Array]:
+    ) -> Generator[CubicSpline]:
```

Comment:
```suggestion
    ) -> Iterable[CubicSpline]:
```
`Generator` has three generic arguments which in this case would be `CubicSpline`, `None`, and `None`. As such, it's better to just use `Iterable`.

Reply:
Wouldn't it be best to be more specific with the return type and use `Generator[CubicSpline, None, None]`?

Reply:
It's semantically equivalent, but it's generally preferred to use `Iterable`.

Reply:
I see. I applied your suggestion and did the same with other `VMobject` methods like `.get_curve_functions()` which also returned a `Generator`.